lib/msal-common/src/client/Client.ts

```diff
@@ -31,18 +31,18 @@ export type ResponseStateInfo = {
 
 /**
  * AuthModule class
- * 
+ *
  * Parent object instance which will construct requests to send to and handle responses from the Microsoft STS using the authorization code flow.
- * 
+ *
  */
-export abstract class AuthModule {
+export abstract class Client {
```

Comment:
We mentioned this already, but this name should be more specific. `InternalClient`, `PrivateClient`, `IPublicClient`? Just some suggestions

Reply:
AuthClient is what I would go with - this could describe either a public or conf client

Reply:
Yes and please note that this is initial cut w.r.t folder structure. @sangonzal  will be making more changes as needed.

Reply:
Cannot be Internal or Public or Private: as all of them are misrepresentative of this base class. `AuthClient` sounds more fitting. :-) @jasonnutter and @DarylThayil ?

Reply:
`AuthClient` is fine. Although it would be great if we were consistent about "Auth" vs "Authentication".

Reply:
We can do `AuthenticationClient` and avoid all `auth` references. Also there will be potentially an `OAuthClient` coming as a base class - @sangonzal feel free to make the changes or let me know today and I can regarding `naming` only changes.

Reply:
> `AuthClient` is fine. Although it would be great if we were consistent about "Auth" vs "Authentication".

This has always been a point of contention since Auth could mean "Authentication" (AuthN) or "Authorization" (AuthZ) :) Although in this case, since the client is doing both, I'd argue it is OK to name it Auth.

Reply:
That's fine, it's just that sometimes we name things "Auth" and sometimes its "Authentication"  (e.g. AuthenticationParameters), and it would be great it we were consistent.

Reply:
Ah I see, yes I think we will be taking a look at naming as part of the updates to common.

Reply:
I agree. I believe the logic behind is that we do both `Authorization` and `Authentication` and `Auth` is a good compromise for both. 

My proposal is to:
- Try referencing `Authentication` or `Authorization` as appropriate if the distinction is clear
- Avoid `Auth` as much as possible when not needed; eg: For config etc.
- `OAuth` to be used for generic protocol references as that is less confusing; mostly will be limited to the `client` folder; cc @sangonzal